src/logger/log_entry.py

```diff
@@ -0,0 +1,166 @@
+
+        def _make(self, data: dict) -> "TrainArrivalLogEntry":
+            return TrainArrivalLogEntry(**data)
+
+    train_id = IntegerField(null=False)
+    station_id = IntegerField(null=False)
+
+
+class TrainDepartureLogEntry(LogEntry):
+    """A LogEntry that represents the departure of a train from a station."""
+
+    class Schema(LogEntry.Schema):
+        """The marshmallow schema for the TrainDepartureLogEntry model."""
+
+        train_id = marsh.fields.Integer(required=True)
+        station_id = marsh.fields.Integer(required=True)
+
+        def _make(self, data: dict) -> "TrainDepartureLogEntry":
+            return TrainDepartureLogEntry(**data)
+
+    train_id = IntegerField(null=False)
+    station_id = IntegerField(null=False)
+
+
+class CreateFahrstrasseLogEntry(LogEntry):
+    """A LogEntry that represents the creation of a fahrstrasse."""
+
+    class Schema(LogEntry.Schema):
+        """The marshmallow schema for the CreateFahrstrasseLogEntry model."""
+
+        fahrstrasse = marsh.fields.String(required=True)
+
+        def _make(self, data: dict) -> "CreateFahrstrasseLogEntry":
+            return CreateFahrstrasseLogEntry(**data)
+
+    fahrstrasse = CharField(null=False)
+
+
+class RemoveFahrstrasseLogEntry(LogEntry):
+    """A LogEntry that represents the removal of a fahrstrasse."""
+
+    class Schema(LogEntry.Schema):
+        """The marshmallow schema for the RemoveFahrstrasseLogEntry model."""
+
+        fahrstrasse = marsh.fields.String(required=True)
+
+        def _make(self, data: dict) -> "RemoveFahrstrasseLogEntry":
+            return RemoveFahrstrasseLogEntry(**data)
+
+    fahrstrasse = CharField(null=False)
+
+
+class SetSignalLogEntry(LogEntry):
+    """A LogEntry that represents the setting of a signal."""
+
+    class Schema(LogEntry.Schema):
+        """The marshmallow schema for the SetSignalLogEntry model."""
+
+        signal_id = marsh.fields.String(required=True)
+        state_before = marsh.fields.Integer(required=True)
+        state_after = marsh.fields.Integer(required=True)
+
+        def _make(self, data: dict) -> "SetSignalLogEntry":
+            return SetSignalLogEntry(**data)
+
+    signal_id = CharField(null=False)
+    state_before = IntegerField(null=False)
+    state_after = IntegerField(null=False)
+
+
+class InjectFaultLogEntry(LogEntry):
+    """A LogEntry that represents the injection of a fault."""
+
+    class Schema(LogEntry.Schema):
+        """The marshmallow schema for the InjectFaultLogEntry model."""
+
+        injection_id = marsh.fields.Integer(required=True)
+        fault_type = marsh.fields.Integer(required=True)
+        injection_position = marsh.fields.String(required=True)
+        duration = marsh.fields.Integer(required=True)
+
+        def _make(self, data: dict) -> "InjectFaultLogEntry":
+            return InjectFaultLogEntry(**data)
+
+    injection_id = IntegerField(null=False)
+    fault_type = IntegerField(null=False)
```

Comment:
Please discuss with @instant-sky how he identifies a faulttype and injection. Is it an integer, uuid or string 😇 I'm unsure at the moment.
(Please update it in the other classes too 🥳 )

Reply:
Is discussed, will be changed